egs/grid/AVSR/local/dataset_audio.py

```diff
@@ -0,0 +1,120 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script is to load the audio data in GRID.
+The class dataset_audio makes each audio batch data have the same shape.
+"""
+import kaldifeat
+import numpy as np
+import os
+
+import torch
+import torchaudio
+from torch.utils.data import Dataset
+
+
+class dataset_audio(Dataset):
+    def __init__(
+        self,
+        video_path: str,
+        anno_path: str,
+        file_list: str,
+        aud_padding: int,
+        sample_rate: int,
+        feature_dim: int,
+        phase: str,
+    ) -> None:
+        """
+        Args:
+          video_path:
+            The dir path of the visual data.
+          anno_path:
+            The dir path of the texts data.
+          file_list:
+            The file which listing all samples for training or testing.
+          aud_padding:
+            The padding for each audio sample.
+          sample_rate:
+            The sample rate for extracting fbank feature.
+          feature_dim:
+            The dim for fbank feature.
+          phase:
+            "train" or "test".
+        """
+        self.anno_path = anno_path
+        self.aud_padding = aud_padding
+        self.sample_rate = sample_rate
+        self.feature_dim = feature_dim
+        self.phase = phase
+        with open(file_list, "r") as f:
+            self.videos = [
+                os.path.join(video_path, line.strip()) for line in f.readlines()
+            ]
+
+        self.data = []
+        for vid in self.videos:
+            items = vid.split(os.path.sep)
+            aud = (
+                vid.replace("lip", "audio_25k").replace("/video/mpg_6000", "")
+                + ".wav"
+            )
+            self.data.append((aud, items[-4], items[-1]))
+
+    def __getitem__(self, idx):
+        (aud, spk, name) = self.data[idx]
+        aud = self._load_aud(aud)
+        aud = self._padding(aud, self.aud_padding)
+        anno = self._load_anno(
+            os.path.join(self.anno_path, spk, "align", name + ".align")
+        )
+
+        return {
+            "aud": torch.FloatTensor(aud),
+            "txt": anno.upper(),
+        }
+
+    def __len__(self):
+        return len(self.data)
+
+    def _load_aud(self, filename):
+        opts = kaldifeat.FbankOptions()
+        opts.device = "cpu"
+        opts.frame_opts.dither = 0
+        opts.frame_opts.snip_edges = False
+        opts.frame_opts.samp_freq = self.sample_rate
+        opts.mel_opts.num_bins = self.feature_dim
+        fbank = kaldifeat.Fbank(opts)
```

Comment:
Please construct `fbank` in the constructor , i.e., in `__init__` and use `self.fbank` here.